hyde.py

```diff
@@ -20,5 +25,55 @@ def cat_recall_query(user_message, cat):
     # Run a LLM chain with the user message as input
     hypothesis_chain = langchain.chains.LLMChain(prompt=hypothesis_prompt, llm=cat._llm)
     answer = hypothesis_chain(user_message)
-    log(answer, "INFO")
-    return answer["text"]
+    
+    # Save HyDE answer in working memory
+    cat.working_memory[HYDE_ANSWER] = answer["text"]
+    
+    print("------------- HYDE -------------")
+    print(f"user message: {user_message}")
+    print(f"hyde answer: {answer['text']}")
+    
+    return user_message
+
+
+# Calculates the average between the user's message embedding and the Hyde response embedding
+def _calculate_vector_average(config: dict, cat):
+    
+    # If average embedding not exists & hyde answer exists ..
+    if AVERAGE_EMBEDDING not in cat.working_memory.keys() and HYDE_ANSWER in cat.working_memory.keys():
+        
+        # Get user message embedding
+        user_embedding = config['embedding']
+        
+        # Calculate hyde embedding from hyde answer
+        hyde_answer = cat.working_memory[HYDE_ANSWER]
+        hyde_embedding = cat.embedder.embed_query(hyde_answer)
```

Comment:
This should be done in the hook above, e.g. at line 31. The embedding can be stored in the working memory along with or replacing the `HYDE_ANSWER`. 
In this way, here the embedding is always the one of the new user's query